Sources/Server/ServerNetworkLayer.cpp

```diff
@@ -0,0 +1,17 @@
+//
+// Created by Rafał Rzeuski on 09.05.16.
+//
+
+#include "ServerNetworkLayer.h"
+#include <iostream>
+
+int ServerNetworkLayer::connectToNode() {
+    TCPSocket socket;
+    SocketAddress nodeAddress;
+    // No local port binding needed, just connect(...)
+    if(socket.connect(socket.internalDescriptor(), nodeAddress) == -1) {
```

Comment:
This gives runtime error i believe - there's no initialization of socket, neither address.